crates/gitbutler-hunk-dependency/src/workspace.rs

```diff
@@ -0,0 +1,137 @@
+use std::{
+    collections::{HashMap, HashSet},
+    path::{Path, PathBuf},
+};
+
+use gitbutler_stack::StackId;
+use itertools::Itertools;
+
+use crate::{diff::InputDiff, hunk::HunkRange, stack::StackRanges};
+
+#[derive(Debug, Default, PartialEq, Clone)]
+pub struct WorkspaceRanges {
```

Comment:
Kinda related to our IRL conversation earlier... so there is this convention to no "stutter" in names - this is module workspace, name WorkspaceRanges. 
So if you want your API to be consumed as a hierarchy, it would be idiomatic to have `workspace::Ranges` at the consumption side.
I tend to prefer a flatter structure - exporting these types in the root level, and in that case I need the name would be `WorkspaceRanges` to be descriptive. 
This is doing both which is suboptimal. Can rename later but trying to help shape the thinking here a bit